Userland/Applications/SoundPlayer/PlaybackManager.cpp

```diff
@@ -114,10 +114,15 @@ void PlaybackManager::next_buffer()
             return;
         }
 
-        // FIXME: This should handle parsing failures gracefully and show them to the user.
-        auto buffer = m_loader->get_more_samples(m_samples_to_load_per_buffer).release_value();
+        auto buffer_or_error = m_loader->get_more_samples(m_samples_to_load_per_buffer);
+        if (buffer_or_error.is_error()) {
+            warnln("Loader returned error {}", buffer_or_error.error().description);
+            return;
+        }
```

Comment:
This apparently partially undoes the 'fix' that I made earlier in 877be0eb437408e40fb1188318a4382d59f11dc1, or at least it breaks the intention of "stop parsing as early as possible".

While the errors are indeed no longer silent or crashing, the playback is still not stopped properly, causing the MP3 loader to potentially print pages of errors before running into some asserts in the parser code anyways. Note that, for easier testing, I reverted the recent EOF-fix in LibAudio (1caad457efb9edca2c6dec22508963deec67d2fe) to introduce a non-recoverable parsing error.

Meanwhile, AudioServer believes that the client can't keep up properly and starts locking up the system (or the MP3 loader just hogs up everything, I haven't looked into that that closely).

In any case, we should look for ways (or leave it as a FIXME at the very least) on how to stop playback properly, show potential errors to the user, and how to reset the parser properly before seeking somewhere else (since throwing errors when seeking is bad).

Reply:
OK, added a FIXME back in :)

The problem with the crash on error behavior is:
* Right now, the MP3 loader is not calculating seek tables correctly (I think, still haven't quite figured out what is exactly going on) so seeking an MP3 file caused a crash (but playback worked fine)
* In general, some media files are faulty enough that they cause errors on loading/seeking (containing some invalid frames etc) but not faulty enough that they can't be played back.